client/web/src/repo/blob/own/FileOwnershipPanel.tsx

```diff
@@ -60,17 +61,91 @@ export const FileOwnershipPanel: React.FunctionComponent<
+    owners?: OwnerFields[]
+}
+
+const OwnExplanation: React.FunctionComponent<OwnExplanationProps> = ({ owners }) => {
+    const [dismissed, setDismissed] = useTemporarySetting('own.panelExplanationHidden')
+
+    const onDismiss = React.useCallback(() => {
+        setDismissed(true)
+    }, [setDismissed])
+
+    if (dismissed) {
+        return null
+    }
+
+    const ownerSearchPredicate = resolveOwnerSearchPredicate(owners)
+
+    return (
+        <MarketingBlock contentClassName={styles.ownExplanationContainer} wrapperClassName="mb-3">
+            <div className="d-flex align-items-start">
+                <div className="flex-1">
+                    <H3 as={H4} className={styles.ownExplanationTitle}>
+                        Sourcegraph Own Preview
+                    </H3>
+                    <Text className={classNames(styles.ownExplanationContent, 'mb-2')}>
+                        Find code owners from a CODEOWNERS file in this repository, or from your external ownership
+                        tracking system here. The <Link to="/help/own">Own documentation</Link> contains more
+                        information.
+                    </Text>
+                    <Text className={classNames(styles.ownExplanationContent, 'mb-1')}>
+                        Sourcegraph Own also works in search:
+                    </Text>
+                    <Button
+                        variant="secondary"
+                        size="sm"
+                        outline={true}
+                        as={Link}
+                        to={`/search?q=file:has.owner(${ownerSearchPredicate})`}
+                        className="mr-2"
+                    >
+                        <SyntaxHighlightedSearchQuery
+                            query={`file:has.owner(${ownerSearchPredicate})`}
+                            searchPatternType={SearchPatternType.standard}
+                        />
+                    </Button>
+                    <Button variant="secondary" size="sm" as={Link} to="/search?q=select:file.owners" outline={true}>
+                        <SyntaxHighlightedSearchQuery
+                            query="select:file.owners"
+                            searchPatternType={SearchPatternType.standard}
+                        />
+                    </Button>
+                </div>
+                <Button aria-label="Dismiss alert" variant="icon" onClick={onDismiss}>
+                    <Icon aria-hidden={true} svgPath={mdiClose} />
+                </Button>
+            </div>
+        </MarketingBlock>
+    )
+}
+
+const resolveOwnerSearchPredicate = (owners?: OwnerFields[]): string => {
+    if (owners) {
+        for (const owner of owners) {
+            if (owner.__typename === 'Person' && owner.user?.username) {
+                return `@${owner.user.username}`
+            }
+        }
+    }
+    return 'johndoe'
+}
+
+interface OwnerListProps {
+    data?: OwnershipConnectionFields | undefined
```

Comment:
According to [the other handbook](https://www.typescriptlang.org/docs/handbook/2/objects.html#optional-properties) a `?` property is already potentnially `undefined`

```suggestion
    data?: OwnershipConnectionFields
```

Reply:
for some reason this is giving a type error without the explicit undefined, I'll poke at it some more. This is where my typescript knowledge is stretched 😄

Reply:
nevermind! must be something wrong with my local editor!